tests/testthat/test-derive_merged.R

```diff
@@ -0,0 +1,292 @@
+                               mode = "last",
+                               match_flag = matched)
+  expected <- adsl %>% mutate(WEIGHTBL = c(68, 88, 55, NA),
+                              matched = c(TRUE, TRUE, TRUE, NA))
+
+  expect_dfs_equal(base = expected,
+                   compare = actual,
+                   keys = c("USUBJID"))
+})
+
+## derive_vars_merged: error if variable in both datasets ----
+test_that("derive_vars_merged: error if variable in both datasets", {
+  expect_error(derive_vars_merged(advs,
+                                  dataset_add = adsl,
+                                  by_vars = vars(USUBJID)),
+               regexp = "")
+})
+
+# derive_vars_merged_dt ----
+## derive_vars_merged_dt: merge first date ----
+test_that("derive_vars_merged_dt: merge first date", {
+  actual <- derive_vars_merged_dt(
+    adsl,
+    dataset_add = ex,
+    order = vars(TRTSDT),
+    by_vars = vars(STUDYID, USUBJID),
+    dtc = EXSTDTC,
+    new_vars_prefix = "TRTS",
+    mode = "first"
+  )
+  expected <-
+    adsl %>% mutate(TRTSDT = ymd(c(
+      "2020-12-07", "2021-01-12", "2021-03-02", NA
+    )),
+    TRTSDTF = NA_character_)
+
+  expect_dfs_equal(base = expected,
+                   compare = actual,
+                   keys = c("USUBJID"))
+})
+
+# derive_vars_merged_dtm ----
+## derive_vars_merged_dtm: merge first date ----
+test_that("derive_vars_merged_dt: merge first date", {
+  actual <- derive_vars_merged_dtm(
+    adsl,
+    dataset_add = ex,
+    order = vars(TRTSDTM),
+    by_vars = vars(STUDYID, USUBJID),
+    dtc = EXSTDTC,
+    new_vars_prefix = "TRTS",
+    time_imputation = "first",
+    mode = "first"
+  )
+  expected <-
+    adsl %>% mutate(TRTSDTM = as_iso_dtm(ymd_hms(
+      c(
+        "2020-12-07T00:00:00",
+        "2021-01-12T12:00:00",
+        "2021-03-02T00:00:00",
+        NA
+      )
+    )),
+    TRTSTMF = c("H", NA, "H", NA))
+
+  expect_dfs_equal(base = expected,
+                   compare = actual,
+                   keys = c("USUBJID"))
+})
+
+# derive_var_merged_cat ----
+## derive_var_merged_cat: merge categorized variable ----
+test_that("derive_vars_merged_cat: merge categorized variable", {
+  get_region <- function(x) {
+    if_else(x %in% c("AUT", "NOR"), "EUROPE", "AFRICA")
+  }
+
+  actual <- derive_var_merged_cat(
+    advs,
+    dataset_add = adsl,
+    by_vars = vars(USUBJID),
+    new_var = REGION,
+    source_var = COUNTRY,
+    cat_fun = get_region
+  )
+
+  expected <- left_join(advs, select(adsl, USUBJID, COUNTRY), by = "USUBJID") %>%
+    mutate(REGION = get_region(COUNTRY)) %>%
+    select(-COUNTRY)
+
+
+  expect_dfs_equal(base = expected,
+                   compare = actual,
+                   keys = c("USUBJID", "AVISIT"))
+})
+
+## derive_var_merged_cat: merge categorized variable and define value for non-matched by groups ----
+test_that("derive_vars_merged_cat: merge categorized variable and define value for non-matched by groups", {
```

Comment:
lint fails at this line, in case you missed that :)

Reply:
Thanks, I'll fix it together with the replacement of the internal `left_join()` calls.